packages/block-library/src/image/edit.js

```diff
@@ -85,6 +92,86 @@ function hasDefaultSize( image, defaultSize ) {
 	);
 }
 
+/**
+ * Checks if a media attachment object has been "destroyed",
+ * that is, removed from the media library. The core Media Library
+ * adds a `destroyed` property to a deleted attachment object in the media collection.
+ *
+ * @param {number} id The attachment id.
+ *
+ * @return {boolean} Whether the image has been destroyed.
+ */
+export function isMediaDestroyed( id ) {
+	const attachment = window?.wp?.media?.attachment( id ) || {};
+	return attachment.destroyed;
+}
+
+/**
+ * A placeholder component that displays if an image has been deleted or cannot be loaded.
+ *
+ * @param {Object} props Component props.
+ * @param {number} props.id The attachment id.
+ * @param {string} props.url Image url.
+ * @param {Function} props.onClear Clears the image attributes.
+ * @param {Function} props.onSelect Callback for <MediaReplaceFlow /> when an image is selected.
+ * @param {Function} props.onSelectURL Callback for <MediaReplaceFlow /> when an image URL is selected.
+ * @param {Function} props.onUploadError Callback for <MediaReplaceFlow /> when an upload fails.
+ * @param {Function} props.onCloseModal Callback for <MediaReplaceFlow /> when the media library overlay is closed.
+ *
+ * @return {boolean} Whether the image has been destroyed.
+ */
```

Comment:
The linter complains to me about misaligned JSDoc comments.

<details>
<summary>Click to expand suggestion</summary>

```suggestion
/**
 * A placeholder component that displays if an image has been deleted or cannot be loaded.
 *
 * @param {Object}   props               Component props.
 * @param {number}   props.id            The attachment id.
 * @param {string}   props.url           Image url.
 * @param {Function} props.onClear       Clears the image attributes.
 * @param {Function} props.onSelect      Callback for <MediaReplaceFlow /> when an image is selected.
 * @param {Function} props.onSelectURL   Callback for <MediaReplaceFlow /> when an image URL is selected.
 * @param {Function} props.onUploadError Callback for <MediaReplaceFlow /> when an upload fails.
 * @param {Function} props.onCloseModal  Callback for <MediaReplaceFlow /> when the media library overlay is closed.
 *
 * @return {boolean} Whether the image has been destroyed.
 */
```

</details>